src/vm/hints/secp/field_utils.rs

```diff
@@ -72,18 +66,9 @@ pub fn reduce(
     ids: &HashMap<String, BigInt>,
     hint_ap_tracking: Option<&ApTracking>,
 ) -> Result<(), VirtualMachineError> {
-    let x_reloc = get_relocatable_from_var_name("x", ids, vm, hint_ap_tracking)?;
-
-    let x_d0 = get_integer_from_relocatable_plus_offset(&x_reloc, 0, vm)?;
-    let x_d1 = get_integer_from_relocatable_plus_offset(&x_reloc, 1, vm)?;
-    let x_d2 = get_integer_from_relocatable_plus_offset(&x_reloc, 2, vm)?;
-
-    //SECP_P = 2**256 - 2**32 - 2**9 - 2**8 - 2**7 - 2**6 - 2**4 - 1
-    let sec_p = bigint_str!(
-        b"115792089237316195423570985008687907853269984665640564039457584007908834671663"
-    );
+    let value = pack_from_var_name("x", ids, vm, hint_ap_tracking)?.mod_floor(&*SECP_P);
```

Comment:
This looks weird. If we have a reference and we pass a reference, why do we need to dereference first?

Reply:
That function converts a struct to a single BigInt value. In that case x is actually a struct with the values x.d0, x.d1 and x.d2

Reply:
I mean the `&*SECP_P`. Wouldn't it suffice to do `SECP_P`?

Reply:
Nvm: https://github.com/lambdaclass/cleopatra_cairo/pull/333/commits/8008c87b35849d56aa208930c471b2b5290e54fc